server/reload.go

```diff
@@ -236,6 +281,133 @@ func (r *routesOption) Apply(server *Server) {
 	server.Noticef("Reloaded: cluster routes")
 }
 
+// maxConnOption implements the option interface for the `max_connections`
+// setting.
+type maxConnOption struct {
+	noopOption
+	newValue int
+}
+
+// Apply the max connections change by closing random connections til we are
+// below the limit if necessary.
+func (m *maxConnOption) Apply(server *Server) {
+	server.mu.Lock()
+	var (
+		clients = make([]*client, len(server.clients))
+		i       = 0
+	)
+	for _, client := range server.clients {
```

Comment:
It may be useful to add a comment to the effect that map iteration is where the randomness is introduced - that feature of go may not be readily apparent to future contributors.